### PR TITLE
Add docs for getting the fields list

### PR DESCRIPTION
**Pull request by nickvergessen**
Code: https://github.com/nextcloud/server/blob/5c3551cf2f57b2ce72f2503e731f10cec5355782/apps/provisioning_api/lib/Controller/UsersController.php#L522-L546
Route: https://github.com/nextcloud/server/blob/fe9c46e595cba2d8e52db13cf15848a1f9d6f141/apps/provisioning_api/appinfo/routes.php#L55

### CODE REVIEW

File: admin_manual/configuration_user/instruction_set_for_users.rst [hunk 1]
```diff
@@ -155,9 +155,10 @@ XML output
 Edit data of a single user
 --------------------------
 
-Edits attributes related to a user. Users are able to edit email, displayname 
-and password; admins can also edit the quota value. Authentication is done by 
-sending a Basic HTTP Authorization header.
+Edits attributes related to a user. Users are able to edit email, displayname
+and password; admins can also edit the quota value. Further restrictions may apply,
+check the :ref:`Editable data fields list <_editable_field_list>`. Authentication
```
Comment (ChristophWurst) [inline admin_manual/configuration_user/instruction_set_for_users.rst:160]: does the link work? the github action annotation says no

Comment (nickvergessen) [reply admin_manual/configuration_user/instruction_set_for_users.rst]: Right, whatever went wrong there... fixed now. Also built locally to confirm

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Was added with nextcloud/server#8000 in 14. So backporting down to 19.

**Pull request merged**